pkg/apis/networking/v1/defaults.go

```diff
@@ -33,3 +33,15 @@ func SetDefaults_NetworkPolicyPort(obj *networkingv1.NetworkPolicyPort) {
 		obj.Protocol = &proto
 	}
 }
+
+func SetDefaults_NetworkPolicySpec(obj *networkingv1.NetworkPolicySpec) {
+	if obj.PolicyType == nil {
+		if obj.Ingress != nil && obj.Egress != nil {
+			obj.PolicyType = []string{"ingress", "egress"}
```

Comment:
You'll need to update things throughout for Casey's suggestion:

    obj.PolicyTypes = []PolicyType{PolicyTypeIngress, PolicyTypeEgress}